Lib/_weakrefset.py

```diff
@@ -106,12 +106,20 @@ def pop(self):
     def remove(self, item):
         if self._pending_removals:
             self._commit_removals()
-        self.data.remove(ref(item))
+        try:
+            wr = ref(item)
+        except TypeError:
+            return KeyError(item)
```

Comment:
Yes I've noticed the tests explicitly check that `discard` and `remove` raise for such types ... I'll patch those tests tomorrow (hopefully!).

(Also, that `return` should have been a `raise`, I've force pushed that so your change has gone outdated)